ringvax/summary.py

```diff
@@ -26,6 +26,17 @@
 assert set(infection_schema.keys()) == Simulation.PROPERTIES
 
 
+def frac(num, denom) -> float:
```

Comment:
This is the behavior of [`numpy.divide`](https://numpy.org/doc/2.1/reference/generated/numpy.divide.html), so we could use that?

It throws a user warning, but you can toggle that off for divide by zero warnings with https://numpy.org/doc/2.1/reference/generated/numpy.seterr.html#numpy.seterr

Reply:
It's also funny to me how much I would prefer
```python
if num == 0.0 and denom == 0.0:
  return nan
elif num > 0.0 and denom == 0.0:
  etc.
```
What/when/where I developed that preference?

Reply:
Looks like we can locally turn off the warnings with `@np.errstate` (otherwise we'd risk other things, like logs of negative numbers not warning elsewhere) so I'm deprecating `frac` in favor of `divide` plus a decorator on `summarize_detections()`